message_ix_models/testing/__init__.py

```diff
@@ -351,3 +353,72 @@ def not_ci(reason=None, action="skip"):
     """
     action = "skipif" if action == "skip" else action
     return getattr(pytest.mark, action)(condition=GHA, reason=reason)
+
+
+def unpack_snapshot_data(context: Context, snapshot_id: int):
+    """Already-unpacked data for a snapshot.
+
+    This copies the .csv.gz files from message_ix_models/data/test/… to the directory
+    where they *would* be unpacked by .model.snapshot._unpack. This causes the code to
+    skip unpacking them, which can be very slow.
+    """
+    if snapshot_id != 0 and snapshot_id != 1:
+        log.info(f"No unpacked data for snapshot {snapshot_id}")
+        return
+
+    dest = context.get_cache_path(
+        f"snapshot-{snapshot_id}",
+        "MESSAGEix-GLOBIOM_1.1_R11_no-policy_baseline",
+    )
+    log.debug(f"{dest = }")
+
+    snapshot_data_path = util.package_data_path(
+        "test",
+        f"snapshot-{snapshot_id}",
+        "MESSAGEix-GLOBIOM_1.1_R11_no-policy_baseline",
+    )
+    log.debug(f"{snapshot_data_path = }")
+
+    shutil.copytree(snapshot_data_path, dest, dirs_exist_ok=True)
+
+
+@pytest.fixture(
+    scope="session",
+    params=[
+        int(k.split("-")[1]) for k in util.pooch.SOURCE if k.startswith("snapshot")
+    ],
+)
+def load_snapshot(request, session_context, solved: bool = False):
+    snapshot_id: int = request.param
+    assert snapshot_id is not None
+    unpack_snapshot_data(context=session_context, snapshot_id=snapshot_id)
+    model_name = "MESSAGEix-GLOBIOM_1.1_R11_no-policy"
+    scenario_name = f"baseline_v{snapshot_id}"
+    mp = session_context.get_platform()
+
+    try:
```

Comment:
Maybe a comment here:
```py
# The following code roughly parallels bare_res()
```
This could be a cue that if we update one, the other might also need updating.

We might also try to merge the two functions, but not in this PR.